sdk/include/platform/concepts/entropy.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+#include <array>
+#include <concepts>
+#include <cstdint>
+
+/**
+ * Concept for an Ethernet adaptor.
+ */
+template<typename T>
+concept IsEntropySource = requires(T source)
+{
+	/**
+	 * Must export a flag indicating whether this is a cryptographically
+	 * secure random number.
+	 */
+	{
+		T::IsSecure
+		} -> std::convertible_to<const bool>;
+
+	/**
+	 * Must return a random number.  All bits of the value type are assumed to
+	 * be random, this should use a narrower type if it guarantees less
+	 * entropy.
```

Comment:
Requiring distillation in the application-facing source is probably fine, but I'm not sure it's worth it if one starts stacking/pooling sources a la Fortuna.

Reply:
I think that real silicon will be likely to come with a decent hardware random number generator (entropy source fed through some running hash construct) and so I don't expect to need Fortuna long term.